src/main/kotlin/io/bazel/kotlin/plugin/jdeps/k2/JdepsK2Utils.kt

```diff
@@ -25,11 +39,16 @@ internal fun DeserializedContainerSource.classId(): ClassId? =
   }
 
 internal fun SourceElement.binaryClass(): String? =
-  when (this) {
-    is KotlinJvmBinarySourceElement -> binaryClass.location
-    is JvmPackagePartSource -> this.knownJvmBinaryClass?.location
-    is JavaBinarySourceElement -> this.javaClass.virtualFile.path
-    else -> null
+  if (this is KotlinJvmBinarySourceElement) {
+    binaryClass.location
+  } else if (this is JvmPackagePartSource) {
+    this.knownJvmBinaryClass?.location
+  } else if (RefCache.jbseClass != null && RefCache.jbseClass!!.isInstance(this)) {
+    val jClass = RefCache.jbseGetJavaClassMethod!!.invoke(this)
+    val virtualFile = jClass!!.javaClass.getMethod("getVirtualFile").invoke(jClass)
+    virtualFile.javaClass.getMethod("getPath").invoke(virtualFile) as? String
```

Comment:
Using `!!` here since we are using reflection and probably want to fail-fast if the underlying methods change somehow.